docs/src/format/table/index/vector/index.md

```diff
@@ -0,0 +1,265 @@
+
+| Column          | Type   | Nullable | Description                                  |
+|-----------------|--------|----------|----------------------------------------------|
+| `__flat_marker` | uint64 | false    | Marker field for FLAT index (no actual data) |
+
+##### HNSW
+
+HNSW (Hierarchical Navigable Small World) indices provide fast approximate search through a multi-level graph structure. This stores the HNSW graph with the following schema:
+
+| Column        | Type          | Nullable | Description            |
+|---------------|---------------|----------|------------------------|
+| `__vector_id` | uint64        | false    | Vector identifier      |
+| `__neighbors` | list<uint32>  | false    | Neighbor node IDs      |
+| `_distance`   | list<float32> | false    | Distances to neighbors |
+
+!!! note
+    HNSW consists of multiple levels, and all levels must be written in order starting from level 0.
+
+### Auxiliary File
+
+The auxiliary file is a vector storage for quantized vectors.
+It is stored as `auxiliary.idx` within the index directory.
+
+#### Metadata
+
+The auxiliary file also contains metadata in its schema metadata for vector storage configuration:
+
+##### "distance_type"
+The distance metric used to compute similarity between vectors (e.g., "l2", "cosine", "dot").
+
+##### "lance:ivf"
+
+Similar to the index file's "lance:ivf" but focused on vector storage layout. 
+This doesn't contain the partitions' centroids.
+It's only used for tracking each partition's offset and length in the auxiliary file.
+
+##### "storage_metadata"
+
+Contains quantizer-specific metadata as a list of JSON strings.
+Currently, the list always contains exactly 1 element with the quantizer metadata.
+
+For **Product Quantization (PQ)**:
+
+| JSON Key            | Type  | Description                                                      |
+|---------------------|-------|------------------------------------------------------------------|
+| `codebook_position` | usize | Position of the codebook in the global buffer                    |
+| `nbits`             | u32   | Number of bits per subvector code (e.g., 8 bits = 256 codewords) |
+| `num_sub_vectors`   | usize | Number of subvectors (m)                                         |
+| `dimension`         | usize | Original vector dimension                                        |
+| `transposed`        | bool  | Whether the codebook is stored in transposed layout              |
+
+For **Scalar Quantization (SQ)**:
+
+| JSON Key   | Type       | Description                            |
+|------------|------------|----------------------------------------|
+| `dim`      | usize      | Vector dimension                       |
+| `num_bits` | u16        | Number of bits for quantization        |
+| `bounds`   | Range<f64> | Min/max bounds for scalar quantization |
+
+#### Global Buffer
+
```

Comment:
Can you add an example about how to read / write global buffer?

Reply:
added example in appendix 2